### DISCUSSION TITLE
How to update subscription item to become recurring using swell-node?

### DISCUSSION THREAD

**Discussion by tarikrital** [swellstores/community#561] [Help] [answered]
Welcome,

I went through the docs but I didn't find how to update subscription item? also graphQL doesn't allow to update this property!

Is there a ref I missed?

**Comment by logeshswell** [comment]
Were you able to follow the instructions in this doc under GraphQL option:
https://developers.swell.is/frontend-api/carts#add-an-item-with-a-purchase-option

**Reply by logeshswell** [reply]
<img width="1091" alt="Screenshot 2024-07-29 at 7 18 15 PM" src="https://github.com/user-attachments/assets/6f1b47d8-3574-4034-ba14-36d31c3c7d3c">

**Reply by logeshswell** [reply]
The same instruction applies for swell-node as well, like you can set `purchase_option.type=subscription`

**Reply by tarikrital** [reply]
> Were you able to follow the instructions in this doc under GraphQL option: https://developers.swell.is/frontend-api/carts#add-an-item-with-a-purchase-option

I checked it but recurring not offered!

Can you share a code of recurring option when updating an item using graphQL?

**Reply by logeshswell** [reply]
recurring is nothing but, adding an item with "subscription" purchase type.
Please correct me if I get it wrong.

**Reply by tarikrital** [reply]
>Please correct me if I get it wrong.

I don't want to add an item to a cart but I want to add item to invoice items of a subscription!

![image](https://github.com/user-attachments/assets/3f2f5510-78bc-4742-968a-017142de1425)

In contrast, not available from the current graphQL.

![image](https://github.com/user-attachments/assets/f45d918f-9fa5-4c17-80c8-1a27d547d97e)

**Reply by logeshswell** [reply, marked as answer]
You can open up frontend permission for "items.recurring" under "subscriptions" collection.
https://developers.swell.is/guides/opening-content-permissions

After updating the permissions for the public key, you could pass `recurring: true/false` inside the `input` field.

**Comment by tarikrital** [comment]
Going through node solution how to update the invoice item (items) only of specific subscription? can you share a code because documentation seems not showing recurring prop either?

![image](https://github.com/user-attachments/assets/099dac72-6cb6-4bab-a0ff-830babf148a2)
